### PR TITLE
change to use pytest

### PR DESCRIPTION
**Pull request by davidmort**
change to use pytest to improve the unit test output and to stop including tests in the code coverage stats
## Checklist
  - [x] CHANGELOG.md updated? (if required)

### CODE REVIEW

**Comment by davidmort** [comment]
[Use EQ json-schema's to validate (3)](https://trello.com/c/iwZAuR9P/390-use-eq-json-schemas-to-validate-3)

**Comment by davidmort** [comment]
[sdx-collect - implement py.test](https://trello.com/c/aDUeyYUb/488-sdx-collect-implement-pytest)

File: Makefile [hunk 1]
```diff
@@ -6,5 +6,5 @@ build:
 test:
 	pip3 install -r test_requirements.txt
 	flake8 --exclude ./lib/*
-	python3 -m unittest tests/*.py
+	pytest --cov app
```
Comment (emilioward) [inline Makefile]: pytest -v --cov app will produce a more verbose output of individual files if thats the direction we want to move in. http://pythontesting.net/framework/pytest/pytest-introduction/

File: .travis.yml [hunk 2]
```diff
@@ -7,9 +7,9 @@ before_install:
     - pip install --upgrade pip
 install:
     - make build
+    - pip install -r test_requirements.txt
     - pip install codecov
 script:
-    - make test
-    - coverage run -m unittest tests/*.py
+    - pytest tests/ --cov=app --cov-report html
```
Comment (emilioward) [inline .travis.yml]: `pytest -v tests/ --cov=app --cov-report html` will produce a more verbose output of individual files if thats the direction we want to move in. http://pythontesting.net/framework/pytest/pytest-introduction/

**Pull request merged**